setup.sh

```diff
@@ -127,22 +124,50 @@ function install_vagrant_plugins {
   vagrant_plugin_install vagrant-gatling-rsync "Vagrant Gatling Rsync"
 }
 
+# env_is_defined VARNAME
+#   Checks if a new $SHELL has $VARNAME defined in its environment.
+#   Returns 0 when VARNAME is defined for new shells, 1 otherwise.
+function env_is_defined {
+  local var="$1"
+  if test -z "${var}" ; then
+    echo >&2 "internal error: no var specified for env_is_defined"
+    exit 1
+  fi
+
+  # First unset the $var in a sub-shell, and then spawn a new shell
+  # to see if it gets re-defined from its startup code.
+  local setting=$( unset "${var}" ;
+                   "${SHELL}" -i -c "env | grep \"^${var}=\"" )
+  test -n "${setting}"
+}
+
 function add_environment_variables {
   local readonly env_file=$(get_env_file)
-  local readonly docker_host_export="\n# docker-osx-dev\n$DOCKER_HOST_EXPORT=$DOCKER_HOST"
 
-  if grep -q "^[^#]*$DOCKER_HOST_EXPORT" "$env_file" ; then
-    log_warn "$env_file already contains \"$DOCKER_HOST_EXPORT\", will not overwrite"
+  if env_is_defined 'DOCKER_HOST' ; then
+    log_warn "${SHELL} setup already defines DOCKER_HOST will not overwrite"
```

Comment:
Perhaps mention that $env_file is a good place to look for where `DOCKER_HOST` is defined?